File: app/src/main/java/com/natateam/myevents/db/Event.java
package com.natateam.myevents.db;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by macbook on 07/07/ 15.
 */
public class Event extends RealmObject {
    public static final String ID ="ID";
    @PrimaryKey
    private long id;
    private long dateMS;
    private String date;
    private String time;
    private String title;
    private String desc;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getDateMS() {
        return dateMS;
    }

    public String getDate() {
        return date;
    }

    public String getTime() {
        return time;
    }

    public String getTitle() {
        return title;
    }

    public String getDesc() {
        return desc;
    }

    public void setDateMS(long dateMS) {
        this.dateMS = dateMS;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public Event(long id) {
        this.id = id;
    }

    public Event() {
    }
}

File: app/src/main/java/com/natateam/myevents/db/Event.kt
package com.natateam.myevents.db

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

/**
 * Created by macbook on 07/07/ 15.
 */
open class Event : RealmObject {
    @PrimaryKey
    var id: Long = 0
    var dateMS: Long = 0
    var task_date: String? = null
    var task_time: String? = null
    var title: String? = null
    var desc: String? = null
    var isTask_is_done: Boolean = false
    var task_type: String? = null
    var task_repeat_days: String? = null
    var task_repeat_times: String? = null
    var is_need_alarm:Boolean = true
    var user_id: String? = null

    fun getRepeatDaysArray():List<String>?{
        return task_repeat_days?.split(" ")
    }

    constructor(id: Long) {
        this.id = id
    }

    constructor() {}

    companion object {
        val ID = "EVENT_ID"
    }
}
